crates/provider/src/layers/batch.rs

```diff
@@ -81,6 +81,13 @@ impl Default for CallBatchLayer {
     }
 }
 
+#[allow(dead_code)]
+#[derive(Debug)]
+enum CallResult {
```

Comment:
I think we can keep using `IMulticall3::Result` and just set the `success` to true always when we're not calling the contract